lisa/sut_orchestrator/baremetal/platform_.py

```diff
@@ -52,18 +52,20 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
         self.key_loader_factory = Factory[KeyLoader](KeyLoader)
         self.source_factory = Factory[Source](Source)
         self.build_factory = Factory[Build](Build)
-
-    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
-        return self._configure_node_capabilities(environment, log)
-
-    def _deploy_environment(self, environment: Environment, log: Logger) -> None:
         # currently only support one cluster
         assert self._baremetal_runbook.cluster, "no cluster is specified in the runbook"
-        cluster_instance = self._baremetal_runbook.cluster[0]
+        self._cluster_runbook = self._baremetal_runbook.cluster[0]
+        self.cluster = self.cluster_factory.create_by_runbook(self._cluster_runbook)
 
-        self.cluster = self.cluster_factory.create_by_runbook(cluster_instance)
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
         assert self.cluster.runbook.client, "no client is specified in the runbook"
+        if self.cluster.runbook.client[0].capabilities:
+            client_capabilities = self.cluster.runbook.client[0].capabilities
+        else:
+            client_capabilities = self.cluster.get_client_capabilities()
```

Comment:
Pass in the client object, when calling get_client_capabilities.

Reply:
update